syntax = "proto3";
package prof;

service prof {
    rpc getProf(profRequest) returns (profResponse);
    rpc getProfList(profListRequest) returns (profListResponse);
}

message profRequest {
    string name = 1;
}

message profResponse {
    Professor prof = 1;
}

message profListRequest {
}

message profListResponse {
    repeated Professor profs = 1;
}

message Professor {
    string name = 1;
    string rating = 2;
    string wouldTakeAgain = 3;
    string levelOfDifficulty = 4;
    string topTags = 5;
    string reviews = 6;
    string numReviews = 7;
}